### ISSUE TITLE
Craft 4 - Blocks aren't returned in GraphQL response with live preview tokens until edited

### ISSUE THREAD

**Issue by weotch** [spicywebau/craft-neo#601]
#### Description

We use Craft with Neo headlessly.  We're running into an issue with Craft 4's live preview where, once you start editing the page, **only** the Neo blocks _that have been edited_ are returned in our graphql queries.  Vanilla matrix blocks are unaffected.  Here's a video of the behavior we're seeing:

https://user-images.githubusercontent.com/77567/173856825-847eecfd-7995-49b6-b260-ca365384ae66.mov

#### Steps to reproduce

To support live previews, like shown in this video, we have a simple bit of JS running in the control panel that tells our app (running in the preview iframe) to refresh it's data when content changes in the preview:

```js
Garnish.on(Craft.Preview, 'beforeUpdateIframe', function(event) {
  if (event.refresh) return
  event.target.$iframe[0].contentWindow.postMessage('preview:change', '*')
})
```

We read the `token` for the preview from the window query params and include it in the GQL query.  In other words, I'm POSTing to an endpoint like `http://craft.test/api?token=2XQ_oUMnLsjz3lDCxmo96YfeWB52ieHG`. Here's an example of the gql query we're running:

```gql
query getTower($uri: [String]) {
  entry(section: "towers", uri: $uri) {
    ... on towers_default_Entry {
      title

      # Neo Blocks
      blocks {
        __typename
        ... on blocks_simpleMarquee_BlockType {
          id
          copy: fullWysiwyg
        }
      }

      # Vanilla Craft blocks
      matrixBlocks {
        __typename
        ... on matrixBlocks_copy_BlockType {
          id
          content: copy
        }
      }
    }
  }
}
```

When you see the Neo block disappear in that video, they are missing from the GQL response.  For instance, after I edited the entry title, the GQL response was like:

```json
{
  "data": {
    "entry": {
      "title": "Homepage Test Edited",
      "blocks": [],
      "matrixBlocks": [
        {
          "__typename": "matrixBlocks_copy_BlockType",
          "id": "71",
          "content": "<p class=\"body\">Matrix block 1</p>"
        }
      ]
    }
  }
}
```

Then when I edited the first Neo block, the response was:

```json
{
  "data": {
    "entry": {
      "title": "Homepage Test Edited",
      "blocks": [
        {
          "__typename": "blocks_simpleMarquee_BlockType",
          "id": "166",
          "copy": "<h1 class=\"style-h1\">Neo block 1 shows up</h1>"
        }
      ],
      "matrixBlocks": [
        {
          "__typename": "matrixBlocks_copy_BlockType",
          "id": "71",
          "content": "<p class=\"body\">Matrix block 1</p>"
        }
      ]
    }
  }
}
```

#### Other information
- Neo version: `3.1.0`
- Craft version: `Craft Pro 4.0.4`
- Is eager-loading used?: Not sure, we didn't _explicitly_ enable it

**Comment by ttempleton** [comment]
Thanks for reporting this. It turned out to affect eager loading in general in preview mode (which affected GraphQL since Craft uses eager loading for GraphQL queries) and should now be fixed in Neo 3.1.1.

**Comment by weotch** [comment]
Yup, that fixed it, thanks!
